src/Permissions.ts

```diff
@@ -53,6 +57,46 @@ export class Permissions {
     return canI;
   }
 
+  /**
+   * Clarifies whether a given user can perform an action for a given K8sCluster.
+   * @param action - `Actions` enum value
+   * @param forSpace - Space object that must contain `{ teams: Team[] }`
+   * @param forK8sCluster - K8sCluster
+   * @param forUserId - string userId
+   * @returns boolean
+   * @throws "Could not get role for space with no teams" exception
+   */
+  canK8sCluster(action: K8sClusterActions, forSpace: Space, forK8sCluster: K8sCluster, forUserId: string) {
+    let canI = false;
+
+    switch (action) {
+      // Admin, Owner or K8sCluster creator can delete it
+      case K8sClusterActions.DeleteK8sCluster: {
+        const isOwnerAdmin = [Roles.Owner, Roles.Admin].includes(this.getRole(forSpace, forUserId));
+        canI = isOwnerAdmin || forK8sCluster.createdById === forUserId;
+        break;
+      }
+
+      default:
+        throw new Error(`Unknown action ${action}`);
+    }
+
+    return canI;
+  }
+
+  /**
+   * DEPRECATED. WILL BE REMOVED.
```

Comment:
Add this so editor like VSCode will warns https://tsdoc.org/pages/tags/deprecated/

Reply:
Thanks, added